src/H5system.c

```diff
@@ -1350,3 +1350,42 @@ H5_get_option(int argc, const char *const *argv, const char *opts, const struct
     /* return the current flag character found */
     return optchar;
 }
+
+/*-------------------------------------------------------------------------
+ * Function:    H5strcasestr
+ *
+ * Purpose:     Find the first occurrence of the substring needle in the
+ *              string haystack ignoring case.
+ *
+ * Return:      Success:  Pointer to the beginning of the located substring
+ *
+ *              Failure:  NULL
+ *
+ * Programmer:  Frank Berghaus
+ *              Sep 2022
```

Comment:
Although it's still common in the library, we're not adding author information to new code. Existing author comments will eventually be removed.